src/test/java/org/jenkinsci/plugins/durabletask/BourneShellScriptTest.java

```diff
@@ -281,85 +415,73 @@ public void smokeTest() throws Exception {
-        c.cleanup(dockerWS);
+    private boolean noZombies() throws InterruptedException {
+        ByteArrayOutputStream baos = null;
         do {
             Thread.sleep(1000);
             baos = new ByteArrayOutputStream();
             try {
-                assertEquals(0, dockerLauncher.launch().cmds("ps", "-e", "-o", "pid,stat,comm").stdout(new TeeOutputStream(baos, System.out)).join());
+                assertEquals(0, launcher.launch().cmds("ps", "-e", "-o", "pid,ppid,stat,comm").stdout(new TeeOutputStream(baos, System.out)).join());
             } catch (IOException x) { // no ps? forget this check
                 System.err.println(x);
                 break;
             }
         } while (baos.toString().contains(" sleep "));
-        assertThat("no zombies running", baos.toString(), not(containsString(" Z ")));
-        s.toComputer().disconnect(new OfflineCause.UserCause(null, null));
+        return !baos.toString().contains(" Z ");
+    }
+
+    // to assist with debugging
+    private String psOut() throws InterruptedException {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try {
+            launcher.launch().cmds("ps", "-ef").stdout(new TeeOutputStream(baos, System.out)).join();
+        } catch (IOException x) { // no ps? skip
+            System.err.println(x);
+        }
+        return baos.toString();
     }
 
-    @Test public void runWithCommandLauncher() throws Exception {
-        assumeTrue("Docker required for this test", new Docker().isAvailable());
-        runOnDocker(new DumbSlave("docker", "/home/jenkins/agent", new SimpleCommandLauncher("docker run -i --rm --name agent jenkinsci/slave:3.7-1 java -jar /usr/share/jenkins/slave.jar")));
+    private boolean simpleOrTini() {
+        if (platform == TestPlatform.SIMPLE || platform == TestPlatform.TINI) {
+            return true;
+        } else {
+            return false;
+        }
     }
 
-    @Test public void runWithTiniCommandLauncher() throws Exception {
-        assumeTrue("Docker required for this test", new Docker().isAvailable());
-        runOnDocker(new DumbSlave("docker", "/home/jenkins/agent", new SimpleCommandLauncher("docker run -i --rm --name agent --init jenkinsci/slave:3.7-1 java -jar /usr/share/jenkins/slave.jar")));
+    private DumbSlave createDockerSlave(DockerContainer container, String javaPath) throws hudson.model.Descriptor.FormException, IOException {
+        return new DumbSlave("docker", "/home/test", new SSHLauncher(container.ipBound(22), container.port(22), "test", "test", "", "", javaPath, null, null));
     }
 
+    private void awaitCompletion(Controller c) throws IOException, InterruptedException {
+        while (c.exitStatus(ws, launcher, listener) == null) {
+            Thread.sleep(100);
+        }
+    }
+
+    private void setGlobalInterpreter(String interpreter) {
+        ExtensionList.lookupSingleton(Shell.DescriptorImpl.class).setShell(interpreter);
```

Comment:
nit: I'd delete this single-line method and inline the body everywhere it was called.